GradeCenter.Services/Schools/SchoolService.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using GradeCenter.Data.Models;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using GradeCenter.Data;
+using GradeCenter.Data.Models.Account;
+using System.Collections;
+
+namespace GradeCenter.Services.Schools
+{
+    public class SchoolService : ISchoolService
+    {
+        private readonly GradeCenterContext _db;
+        public SchoolService(GradeCenterContext gradeCenterContext)
+        {
+            _db = gradeCenterContext;
+        }
+
+        /// <summary>
+        /// Gets all existing school entries in the database.
+        /// </summary>
+        /// <returns></returns>
+        public IEnumerable<School> Read()
+        {
+            return _db.Schools.Where(school => school.IsActive)
+                .ToList();
+        }
+
+        /// <summary>
+        /// Takes an number as a paramater
+        /// and returns the associated 
+        /// School entity instance.
+        /// </summary>
+        /// <param name="number"></param>        
+        /// <returns></returns>
+        public School? GetSchoolByNumber(int number)
```

Comment:
This "Number" property shouldn't be added, use "Name" or "Id" to get the school. Best scenario is to have both.